aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PersistentMutationOutbox.java

```diff
@@ -414,5 +425,25 @@ private Completable unexpectedMutationScenario() {
                 "Please report at https://github.com/aws-amplify/amplify-android/issues."
             ));
         }
+
+        private PendingMutation<Model> merge(PendingMutation<T> incoming, PendingMutation<T> existing)
```

Comment:
Maybe this could be a utility method in the `PendingMutation` class?  I'm just thinking we should try to avoid `PersistentMutationOutbox` from getting too long.